src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -637,8 +637,15 @@ def build_binary(
                 if post_action not in post_actions:
                     post_actions.append(copy.copy(post_action))
         else:
-            merged_queries = select_left.queries[:-1] + select_right.queries[:-1]
-            post_actions = select_left.post_actions + select_right.post_actions
+            merged_queries = select_left.queries[:-1]
```

Comment:
I think we can merge the code for query merge here for the if else branch, we only need a parameter protection for referenced_ctes

Reply:
referenced_ctes are only enabled when we have `_query_compilation_stage_enabled=True` right. This is the False branch.
What kind of protection are you thinking of exaclty.